src/sentry/integrations/gitlab/search.py

```diff
@@ -27,43 +30,51 @@ def installation_class(self):
         return GitlabIntegration
 
     def handle_search_issues(self, installation: T, query: str, repo: str | None) -> Response:
-        assert repo
+        with self.record_event(
+            SourceCodeSearchEndpointInteractionType.HANDLE_SEARCH_ISSUES
+        ).capture() as lifecycle:
+            assert repo
 
-        full_query: str | None = query
+            full_query: str | None = query
 
-        try:
-            iids = [int(query)]
-            full_query = None
-        except ValueError:
-            iids = None
+            try:
+                iids = [int(query)]
+                full_query = None
+            except ValueError:
+                iids = None
 
-        try:
-            response = installation.search_issues(query=full_query, project_id=repo, iids=iids)
-        except ApiError as e:
-            return Response({"detail": str(e)}, status=400)
+            try:
+                response = installation.search_issues(query=full_query, project_id=repo, iids=iids)
+            except ApiError as e:
+                lifecycle.record_failure(str(e))
```

Comment:
You can just pass the exception through and we'll log the full stack trace unless you think that level of detail is unnecessary.